### ISSUE TITLE
MEFE API - user creation

### ISSUE THREAD

**Issue by kaihendry** [unee-t/lambda2sns#10]
Related to #9, Example payload:

```
{
	"actionType": "CREATE_USER",
	"creatorId": "R4vBD6BZRCNx8JwnM",
	"emailAddress": "naor0example.com",
	"firstName": "Naor",
	"lastName": "Biton",
	"phoneNumber": "28282828"
}
```

To proceed need:
* JSON_OBJECT from Lambda example
* SQL that needs to called on response

**Comment by franck-boullier** [comment]
Once the API has returned a success, the SQL to update the 'upstream' system is

- IF API returns 200

SET @user_creation_request_id= 'userCreationRequestId';
SET @mefe_user_id= 'userId (from_API_response)';
SET @creation_datetime = 'timestamp (from_API_response)';
SET @is_created_by_me = 0;
CALL ut_creation_success_mefe_user_id;

- IF API returns 201

SET @user_creation_request_id= 'userCreationRequestId';
SET @mefe_user_id= 'userId (from_API_response)';
SET @creation_datetime = 'timestamp (from_API_response)';
SET @is_created_by_me = 1;
CALL ut_creation_success_mefe_user_id;

**Comment by franck-boullier** [comment]
# The Test:

In the DEV/Staging environment:
Create a new person in the Unee-T Enterprise database (user email derp@derp.com)

# Expected result:

## If Email does not exist in Mongo
- A user is created in the Mongo DB
- A user is created in the BZ Db
- The record for the created user is updated in the Unee-T Enterprise database

## If Email already exists in Mongo:
- The record for the user is updated in the Unee-T Enterprise database

# Actual result:

- Lambda is fired out of the Unee-T Enterprise Database DEV/Staging
![image](https://user-images.githubusercontent.com/31331637/54727715-39f40280-4bb4-11e9-9363-dc51955f7f09.png)
- User is NOT created in the Mongo DB
- User is NOT created in the BZDb
- Record for the created user is NOT updated in the Unee-T Enterprise database

@kaihendry any idea what might be happening?

**Comment by kaihendry** [comment]
Sorry, somehow missed this. It's not implemented yet. Working on it now.

**Comment by franck-boullier** [comment]
# The Test:

In the DEV/Staging environment:
Create a new person in the Unee-T Enterprise database user 21

# Expected result:

## If Email does not exist in Mongo
- A user is created in the Mongo DB
- A user is created in the BZ Db
- The record for the created user is updated in the Unee-T Enterprise database

## If Email already exists in Mongo:
- The record for the user is updated in the Unee-T Enterprise database

# Actual result:

- Lambda is fired out of the Unee-T Enterprise Database DEV/Staging as expected.
- User is NOT created in the Mongo DB
- User is NOT created in the BZDb
- Record for the created user is NOT updated in the Unee-T Enterprise database

@kaihendry any idea what might be happening?

**Comment by franck-boullier** [comment]
Latest Test worked as expected: user is created in MEFE and BZFE
It is possible to access the MEFE after resetting the password for the user
Well done @kaihendry  and @nbiton 🎉 !!!

**Comment by franck-boullier** [comment]
Re-opening this as there was some unexpected errors:

# The Test:

Done in the DEV/Staging 
- Create a batch of 30 users to create
- Use a SQL script to import these users in the Unee-T enterprise database 
- Check that all the lambdas are fired from the Unee-T enterprise database PASSED - 30 different payloads are sent out.
- Check that all the newly created user have been assigned a MEFE user id FAILED

## Actual result:

Only 14 of the 30 users have a MEFE user ID
16 records have NOT been processed

@kaihendry any idea what might have gone wrong?

**Comment by kaihendry** [comment]
I'm getting 400 errors from the MEFE with `[[object Object]]` as the response. So I need @nbiton to look into this.

**Comment by nbiton** [comment]
Seems to be caused by requests that include email addresses for users who are already BZ users, but not registered to MEFE

**Comment by franck-boullier** [comment]
> Seems to be caused by requests that include email addresses for users who are already BZ users, but not registered to MEFE

That's probably the right explanation
How an we fix that?

**Comment by franck-boullier** [comment]
# The Test:

Done in the DEMO Environment 
- Create a batch of 30 users to create
- Use a SQL script to import these users in the Unee-T enterprise database 
- Check that all the lambdas are fired from the Unee-T enterprise database PASSED - 30 different payloads are sent out.
- Check that all the newly created user have been assigned a MEFE user id FAILED

## Actual result:

NO user have been created

@kaihendry any idea what might have gone wrong?

**Comment by franck-boullier** [comment]
After fixing the issues described in https://github.com/unee-t/bz-database/issues/120

# The Test:

Done in the DEMO Environment 
- Create a batch of 30 users to create
- Use a SQL script to import these users in the Unee-T enterprise database 
- Check that all the lambdas are fired from the Unee-T enterprise database PASSED - 30 different payloads are sent out.
- Check that all the newly created user have been assigned a MEFE user id FAILED

## Actual result:

NO user have been created

@kaihendry any idea what might have gone wrong?

**Comment by kaihendry** [comment]
https://ap-southeast-1.console.aws.amazon.com/cloudwatch/home?region=ap-southeast-1#logEventViewer:group=/aws/lambda/alambda_simple;start=PT5M logs showed there was a timeout due to VPC connectivity which was fixed by 3e752acdc6e27e96e9515f87e092eb3336b9d26d

**Comment by franck-boullier** [comment]
After fixing the timeout issue in AWS 

# The Test:

Done in the DEMO Environment 
- Create a batch of 30 users to create
- Use a SQL script to import these users in the Unee-T enterprise database 
- Check that all the lambdas are fired from the Unee-T enterprise database PASSED - 30 different payloads are sent out.
- Check that all the newly created user have been assigned a MEFE user id PASSED

User creation mechanism now seems to work on the DEV and in the DEMO environment.
